glean-core/android/src/main/java/mozilla/telemetry/glean/private/JweMetricType.kt

```diff
@@ -0,0 +1,172 @@
+) {
+    /**
+     * The public constructor used by automatically generated metrics.
+     */
+    constructor(
+        disabled: Boolean,
+        category: String,
+        lifetime: Lifetime,
+        name: String,
+        sendInPings: List<String>
+    ) : this(handle = 0, disabled = disabled, sendInPings = sendInPings) {
+        val ffiPingsList = StringArray(sendInPings.toTypedArray(), "utf-8")
+        this.handle = LibGleanFFI.INSTANCE.glean_new_jwe_metric(
+            category = category,
+            name = name,
+            send_in_pings = ffiPingsList,
+            send_in_pings_len = sendInPings.size,
+            lifetime = lifetime.ordinal,
+            disabled = disabled.toByte()
+        )
+    }
+
+    /**
+     * Set a JWE value.
+     *
+     * @param value The compact representation of a JWE value.
+     */
+    fun setWithCompactRepr(value: String) {
+        if (disabled) {
+            return
+        }
+
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.launch {
+            LibGleanFFI.INSTANCE.glean_jwe_set_with_compact_repr(this@JweMetricType.handle, value)
+        }
+    }
+
+    /**
+     * Build a JWE value from its elements and set to it.
+     *
+     * @param header A variable-size JWE protected header.
+     * @param key A variable-size encrypted key.
+     *      This can be an empty octet sequence.
+     * @param initVector A fixed-size, 96-bit, base64 encoded Jwe initialization vector.
+     *      If not required by the encryption algorithm, can be an empty octet sequence.
+     * @param cipherText The variable-size base64 encoded cipher text.
+     * @param authTag A fixed-size, 132-bit, base64 encoded authentication tag.
+     *      Can be an empty octet sequence.
+     */
+    fun set(header: String, key: String, initVector: String, cipherText: String, authTag: String) {
+        if (disabled) {
+            return
+        }
+
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.launch {
+            LibGleanFFI.INSTANCE.glean_jwe_set(this@JweMetricType.handle, header, key, initVector, cipherText, authTag)
+        }
+    }
+
+    /**
+     * Tests whether a value is stored for the metric for testing purposes only. This function will
+     * attempt to await the last task (if any) writing to the the metric's storage engine before
+     * returning a value.
+     *
+     * @param pingName represents the name of the ping to retrieve the metric for.
+     *                 Defaults to the first value in `sendInPings`.
+     * @return true if metric value exists, otherwise false
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    @JvmOverloads
+    fun testHasValue(pingName: String = sendInPings.first()): Boolean {
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.assertInTestingMode()
+
+        val res = LibGleanFFI.INSTANCE.glean_jwe_test_has_value(this.handle, pingName)
+        return res.toBoolean()
```

Comment:
```suggestion
        return LibGleanFFI.INSTANCE.glean_jwe_test_has_value(this.handle, pingName).toBoolean()
```

?

Reply:
https://github.com/mozilla/glean/blob/main/glean-core/android/src/main/java/mozilla/telemetry/glean/private/StringMetricType.kt#L91-L92 copy paste.

Reply:
Yeah, not sure why the original does that. Maybe we used to do something else as well